lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -224,7 +223,15 @@ def test_push_status(status_code, message, exit_code):
         )
         assert result.exit_code == exit_code, result.stderr
         assert result.stdout == ""
-        assert result.stderr.strip() == "" if not message else message["message"]
+
+        try:
+            err_msg = "" if not message else message["message"]
+        except TypeError:
+            err_msg = message
+
+        if status_code >= HTTPStatus.BAD_REQUEST:
+            err_msg = f"HTTP Error status: {status_code.value}, message: {err_msg}"
```

Comment:
I'll grant that the numeric values of the `HTTPStatus` codes are all well-defined, but, for the most part, they are not ordered, so applying a relational operator (`>=`) to them seem inappropriate.

Choosing the least of evils, here, I would use a bare `400`, as that _is_ an [architected number](https://httpwg.org/specs/rfc9110.html#status.4xx) (although, it would be better if we had a symbolic constant for it with an _appropriate_ semantic, unlike `BAD_REQUEST`).

Reply:
I opened issue #3374 to deal with the remaining problems and objections. I think I have captured them all.